client/index.jsx

```diff
@@ -1,37 +1,67 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import $ from 'jquery';
+import Items from './Items.jsx';
+import LargePlayer from './LargePlayer.jsx';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       largePlayer: null,
-      mediaRoll: []
+      mediaRoll: [],
+      activeItem: 0
     }
-
+    this.rotateMedia = this.rotateMedia.bind(this)
   }
   componentDidMount() {
     this.loadMedia()
   }
 
-  loadMedia () {
-    console.log('test is working')
+  loadMedia() {
     $.ajax({
       method: 'GET',
       url: '/media/items',
       success: (data) => {
-        console.log(data)
+        this.setState({
+          largePlayer: JSON.parse(data)[0].url,
+          mediaRoll: JSON.parse(data)
+        })
+        // this.rotateMedia()
       },
       error: (err) => {
         console.log('error with ajax loadMedia: ', err)
       }
     })
   }
 
+  rotateMedia(pic) {
+    var array = this.state.mediaRoll
+    this.setState({ largePlayer: array[pic].url, activeItem: pic })
+
+    clearInterval(loopPictures)
+
+    var loopPictures = setInterval(() => {
+      if (this.state.largePlayer === array[array.length - 1].url) {
+        this.setState({ largePlayer: array[2].url, activeItem: 1 })
+      } else {
```

Comment:
Is this array zero-based or one-based?

Reply:
Zero based. When the pictures start rotating, it only rotates to the pictures, when it reaches the end of the thumbnail array, it goes to the first picture, not the first video.